### ISSUE TITLE
files in ignore list should always be ignored

### ISSUE THREAD

**Issue by fonsecas72** [phpmd/phpmd#102]
PHPMD should ignore the file when it's on the ignored list even if the file is the only one in the include list.

This behaviour is similar to other static analisys tools like phpcs-CodeSniffer.

e.g.:
I have a folder named "folderX" with:
file1.php
file2.php

if I use 
``` 
phpmd folderX/ xml myRuleset.xml --exclude "*/folderX/*" 
```

No warnings are triggered and that's correct!

but if I do:
```
phpmd folderX/file1.php xml myRuleset.xml --exclude "*/folderX/*" 
```

It will output warnings! And from my point of view that's not correct, but...

**Comment by marlun** [comment]
I have some problems with this too when trying to integrate phpmd with vim/syntastic where it tests only one file and then my --exclude don't work.

Are there any solutions?

**Comment by jeanCarloMachado** [comment]
I checked the code this is in fact a pdepend issue, I'll try to send a fix soon.

**Comment by manuelpichler** [comment]
PR merged into PDepend, so this will be closed with the next dependency update of PHPMD

**Comment by ravage84** [comment]
Close as https://github.com/phpmd/phpmd/commit/eb7dfeea78ae8d9a09198c7aebbb96d1dd3ecd72 has been merged into 2.4.3
